### ISSUE TITLE
Rancher CLI should support multiline env vars

### ISSUE THREAD

**Issue by nerdicbynature** [rancher/rancher#9305]
**Rancher versions:**
rancher/server: 1.6.2
rancher/agent: 1.2.2

Rancher CLI: 0.6.2

**Docker version: (`docker version`,`docker info` preferred)**
17.03.1-ce

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
RHEL7.3

**Steps to Reproduce:**

Having docker-compose.yml file like this
```
version: '2'
services:
  myservice:
    image: some_image 
    environment:
      SSL_CA_CERT: ${SSL_CA_CERT}
```

I'd like to pass a X509/base64 multiline ssl certificate into a container using Rancher cli. This example works fine if I set an environment variable like this

```
export SSL_CA_CERT="1
2
3
"
```

If I start the container running
`rancher up -d `

execing into the container, "env" correctly shows
```
SSL_CA_CERT=1
2
3
```

However, if I create the following env.txt
```
SSL_CA_CERT="1
2
3
"
```

and deploy the container this time by adding "-e"
`rancher up -d -e env.txt`

execing into the container, "env" **incorrectly** shows:
`SSL_CA_CERT=1`

The same applies to the answer file with "rancher catalog install ... -a answer.txt"

As for the Go code in
_cli/vendor/github.com/rancher/rancher-compose-executor/config/merge.go_
and 
_cli/cmd/stack.go_

it seems that the code just evaluates lines correctly with a key=val pattern:
```
        for scanner.Scan() {
                line := strings.TrimSpace(scanner.Text())
                if line == "" || strings.HasPrefix(line, "#") {
                        continue
                }
                parts := strings.SplitN(line, "=", 2)
                if len(parts) == 1 {
                        answers[parts[0]] = ""
                } else {
                        answers[parts[0]] = parts[1]
                }
        }
```

Any chance that Rancher cli can support multiline vars?

Kind regards,
Andre

**Comment by cjellick** [comment]
initial review done. UX wasn't right based on feedback from @ibuildthecloud and @vincent99. More work needed

**Comment by amy** [comment]
Available on `v1.6.8-rc5`

**Comment by amy** [comment]
Per offline discussion the following UX will be implemented:

To parse multi-line environment files, please suffix the file with with .yaml, .yml, or .json and create the environment file with those respective formats.

All values that are not a number, bool, or string should result in an error.

**Comment by soumyalj** [comment]
Tested using CLI v1.6.4-rc2 on rancher v1.6.8-rc5  Created environment variables files for multiline .json and .yml formats. Created stack using these files and the env variables were displayed correctly in the container shell. Examples below
CLI stack creation command: 
```
./rancher up -d -f multiline.yml -e envfile.json -s multilintest
```

envfile.yml
```
SSL_CA_CERT : |
  1
  2
  3
```
envfile.json
```
{
	"SSL_CA_CERT": "1\n2\n3abc"
}
```

**Comment by ekristen** [comment]
I've tested this as well. Works nicely. Thanks!

**Comment by ekristen** [comment]
Actually this doesn't seem to work for the answers file on `rancher catalog install`, only `rancher up`

**Comment by deniseschannon** [comment]
`rancher catalog install` is available to test with v0.6.4-rc5.

**Comment by soumyalj** [comment]
Tested with CLI v0.6.4-rc5. Created a private catalog repository and launched the catalog using 
```
./rancher catalog install test/multiline-test:v1.0 -a answers.yml
```
Catalog was deployed successfully. The container displayed the environment variable in multiline format. Example below:
```
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
PWD=/
DESC=my test
is
a
filei
23
```
It was tested for answers file in both json and yml format
